### PR TITLE
[SPARK-10198][SQL] Turn off partition verification by default

### PR DESCRIPTION
**Pull request by marmbrus**

### CODE REVIEW

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/QueryPartitionSuite.scala [hunk 1]
```diff
@@ -18,50 +18,54 @@
...
   import ctx.implicits._
-  import ctx.sql
+
+  protected def _sqlContext = ctx
 
   test("SPARK-5068: query data when path doesn't exist"){
-    val testData = ctx.sparkContext.parallelize(
-      (1 to 10).map(i => TestData(i, i.toString))).toDF()
-    testData.registerTempTable("testData")
+    withSQLConf((SQLConf.HIVE_VERIFY_PARTITION_PATH.key, "false")) {
```
Comment (jeanlyn) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/QueryPartitionSuite.scala]: Should be set to `true`?

Comment (marmbrus) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/QueryPartitionSuite.scala]: Ah yes, good catch.

**Comment by marmbrus** [comment]
Failing test passes locally after the config fix.  I'm going to merge so this can be included in the next RC.

File: sql/core/src/main/scala/org/apache/spark/sql/SQLConf.scala [hunk 2]
```diff
@@ -312,7 +312,7 @@ private[spark] object SQLConf {
     doc = "When true, enable filter pushdown for ORC files.")
 
   val HIVE_VERIFY_PARTITION_PATH = booleanConf("spark.sql.hive.verifyPartitionPath",
-    defaultValue = Some(true),
+    defaultValue = Some(false),
```
Comment (CaptionKid) [inline sql/core/src/main/scala/org/apache/spark/sql/SQLConf.scala]: Why this place use false as default value

**Pull request closed without merging**